tests/functional/test_bundle_support.py

```diff
@@ -158,23 +161,46 @@ def test_run_parametrized_action_must_be_runned(sdk_client_fs: ADCMClient):
     ("provider", "on_fail", "was_dict"),
     ("provider", "on_fail", "was_list"),
     ("provider", "on_fail", "was_sequence"),
-    ("host", "on_success", "was_dict"),
-    ("host", "on_success", "was_list"),
-    ("host", "on_success", "was_sequence"),
-    ("host", "on_fail", "was_dict"),
-    ("host", "on_fail", "was_list"),
-    ("host", "on_fail", "was_sequence"),
+    ("cluster", "on_success", "was_dict_new_dsl"),
+    ("cluster", "on_success", "was_list_new_dsl"),
+    ("cluster", "on_success", "was_sequence_new_dsl"),
+    ("cluster", "on_fail", "was_dict_new_dsl"),
+    ("cluster", "on_fail", "was_list_new_dsl"),
+    ("cluster", "on_fail", "was_sequence_new_dsl"),
 ]
 
 
 @pytest.mark.parametrize(("entity", "state", "case"), state_cases)
-def test_load_should_fail_when(sdk_client_fs: ADCMClient, entity, state, case):
+def test_load_should_fail_on_wrong_states(sdk_client_fs: ADCMClient, entity, state, case):
     with allure.step(f"Upload {entity} bundle with {case}"):
         bundle_path = utils.get_data_dir(__file__, "states", entity, state, case)
-        with pytest.raises(coreapi.exceptions.ErrorMessage) as e:
-            sdk_client_fs.upload_from_fs(bundle_path)
-    with allure.step(f"Check if state is {state}"):
-        err.INVALID_OBJECT_DEFINITION.equal(e, state, "should be a <class 'str'>")
+        with catch_failed(Failed, "Bundle was loaded but should fail to load"):
+            with pytest.raises(coreapi.exceptions.ErrorMessage) as e:
+                sdk_client_fs.upload_from_fs(bundle_path)
+    assert_args = [e]
+    # validation messages in new DSL are a bit messy so we check only error code and type
+    if "new_dsl" not in case:
+        assert_args.append(state)
+    with allure.step("Assert that error message is correct"):
+        err.INVALID_OBJECT_DEFINITION.equal(*assert_args)
```

Comment:
Why not be more straightforward:
```
with ...:
    if "new_dsl" not in case:
        err.INVALID_OBJECT_DEFINITION.equal(e)
    else:
        err.INVALID_OBJECT_DEFINITION.equal(e, state)
```
for a lesser nesting level?

Reply:
Fixed